src/components/Observations/hooks/fetchObservations.js

```diff
@@ -45,193 +97,27 @@ const useFetchObservations = ( ): Array<Object> => {
-  const comments = [];
-
-  if ( obs.comments.length > 0 ) {
-    obs.comments.forEach( ( photo ) => {
-      const linkedComment = createCommentForRealm( photo );
-      comments.push( linkedComment );
-    } );
-  }
-  return comments;
-}, [] );
-
-const createObservationForRealm = useCallback( ( obs ) => {
-  const identifications = createLinkedIdentifications( obs );
-  const photos = createLinkedPhotos( obs );
-  const comments = createLinkedComments( obs );
-
-  return {
-    uuid: obs.uuid,
-    commentCount: obs.comment_count || 0,
-    comments,
-    commonName: obs.taxon.preferred_common_name,
-    createdAt: obs.created_at,
-    description: obs.description,
-    identificationCount: obs.identifications.length,
-    identifications,
-    // obs detail on web says geojson coords are preferred over lat/long
-    // https://github.com/inaturalist/inaturalist/blob/df6572008f60845b8ef5972a92a9afbde6f67829/app/webpack/observations/show/ducks/observation.js#L145
-    latitude: obs.geojson.coordinates[1],
-    location: obs.location,
-    longitude: obs.geojson.coordinates[0],
-    photos,
-    placeGuess: obs.place_guess,
-    qualityGrade: obs.quality_grade,
-    taxonRank: obs.taxon.rank,
-    timeObservedAt: obs.time_observed_at,
-    userProfilePhoto: obs.user.icon_url,
-    userLogin: obs.user.login,
-    userPhoto: obs.photos[0].url
-  };
-}, [createLinkedIdentifications, createLinkedPhotos, createLinkedComments] );
-
 const writeToDatabase = useCallback( ( results ) => {
-    if ( results.length === 0 ) {
-      return;
-    }
-    // Everything in the function passed to "realm.write" is a transaction and will
-    // hence succeed or fail together. A transcation is the smallest unit of transfer
-    // in Realm so we want to be mindful of how much we put into one single transaction
-    // and split them up if appropriate (more commonly seen server side). Since clients
-    // may occasionally be online during short time spans we want to increase the probability
-    // of sync participants to successfully sync everything in the transaction, otherwise
-    // no changes propagate and the transaction needs to start over when connectivity allows.
+    if ( results.length === 0 ) { return; }
     const realm = realmRef.current;
     results.forEach( obs => {
-      const newObs = createObservationForRealm( obs );
+      const newObs = Observation.createObservationForRealm( obs );
       realm?.write( ( ) => {
-        const existingObs = realm.objects( "Observation" ).filtered( `uuid = '${obs.uuid}'` );
-        if ( existingObs.length > 0 ) {
+        const existingObs = realm.objectForPrimaryKey( "Observation", obs.uuid );
+        if ( existingObs !== undefined ) {
+          // TODO: modify existing objects when syncing from inatjs
           return;
         }
         realm?.create( "Observation", newObs );
       } );
     } );
-}, [createObservationForRealm] );
+    setLoading( false );
+}, [] );
```

Comment:
Yep, the empty array is used to make sure the `writeToDatabase` function only runs once here, since it's basically telling React that the effect doesn't need to re-run based on any props/state changes. `useCallback` won't run at all without a second argument.